src/Private/Watchdog.ps1

```diff
@@ -0,0 +1,1097 @@
+    $watchdog = $PodeContext.Server.Watchdog.Client
+
+    if ($PodeContext.Server.Watchdog.Client.Status -ne 'Restarting') {
+        # Watchdog client has stopped, updating status to 'Stopped'
+        Set-PodeWatchdogHearthbeatStatus -Status 'Stopped'
+    }
+
+    # Remove the timer associated with the Watchdog client
+    Remove-PodeTimer -Name '__pode_watchdog_client__'
+
+
+
+    # Send the last heartbeat to the watchdog server
+    Send-PodeWatchdogData
+
+    # Disable the Watchdog client
+    $watchdog.Enabled = $false
+
+    # Check if the PipeClient exists and clean up its resources
+    if ($null -ne $watchdog.PipeClient) {
+
+        # Dispose of the PipeReader if it exists
+        if ($null -ne $watchdog.PipeReader) {
+            try {
+                $watchdog.PipeReader.Dispose()
+            }
+            catch {
+                $_ | Write-PodeErrorLog -Level Verbose  # Log any errors during PipeReader disposal
+            }
+        }
+
+        # Dispose of the PipeWriter if it exists
+        if ($null -ne $watchdog.PipeWriter) {
+            try {
+                $watchdog.PipeWriter.Dispose()
+            }
+            catch {
+                $_ | Write-PodeErrorLog -Level Verbose  # Log any errors during PipeWriter disposal
+            }
+        }
+
+        # Disconnect the PipeClient if it is still connected
+        if ($watchdog.PipeClient.IsConnected) {
+            $watchdog.PipeClient.Disconnect()
+        }
+
+        # Dispose of the PipeClient itself
+        try {
+            $watchdog.PipeClient.Dispose()
+        }
+        catch {
+            $_ | Write-PodeErrorLog -Level Verbose  # Log any errors during PipeClient disposal
+        }
+    }
+}
+
+<#
+.SYNOPSIS
+    Sets the status of the Pode Watchdog heartbeat for the client component.
+
+.DESCRIPTION
+    This function updates the status of the Pode Watchdog client heartbeat. It allows you to specify the current state of the Watchdog, such as 'Starting', 'Restarting', 'Running', 'Stopped', etc.
+
+.PARAMETER Status
+    Specifies the new status for the Pode Watchdog client heartbeat. Must be one of the following values:
+    - 'Starting'
+    - 'Restarting'
+    - 'Running'
+    - 'Undefined'
+    - 'Stopping'
+    - 'Stopped'
+    - 'Offline'
+
+.PARAMETER Force
+    Specifies whether to force to send the update of the Pode Watchdog client's heartbeat status to the server.
+
+.EXAMPLE
+    Set-PodeWatchdogHearthbeatStatus -Status 'Running'
+
+    This command sets the Watchdog client's heartbeat status to 'Running'.
+
+.NOTES
+    This function checks if the Pode Watchdog client is enabled before updating the status.
+    This is an internal function and may change in future releases of Pode.
+#>
+function Set-PodeWatchdogHearthbeatStatus {
+    param(
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Starting', 'Restarting', 'Running', 'Undefined', 'Stopping', 'Stopped', 'Offline')]
```

Comment:
Another place I think it would be helpful to have enums and maybe a watchdog class.

Reply:
I’ll try with Enum. But normally are a pain in poweshell

Reply:
I usually have an Enum folder and with a `.ps1` which I dot load or toss at the top when the module is compiled. If it's a function that's only ever used by internal developers then it may not be worth the hassle.